pkg/targetallocator/container_test.go

```diff
@@ -77,3 +79,33 @@ func TestContainerVolumes(t *testing.T) {
 	assert.Len(t, c.VolumeMounts, 1)
 	assert.Equal(t, naming.TAConfigMapVolume(), c.VolumeMounts[0].Name)
 }
+
+func TestContainerResourceRequirements(t *testing.T) {
+	otelcol := v1alpha1.OpenTelemetryCollector{
+		Spec: v1alpha1.OpenTelemetryCollectorSpec{
+			TargetAllocator: v1alpha1.OpenTelemetryTargetAllocator{
+				Resources: corev1.ResourceRequirements{
+					Limits: corev1.ResourceList{
+						corev1.ResourceCPU:    resource.MustParse("100m"),
+						corev1.ResourceMemory: resource.MustParse("128M"),
+					},
+					Requests: corev1.ResourceList{
+						corev1.ResourceCPU:    resource.MustParse("200m"),
+						corev1.ResourceMemory: resource.MustParse("256M"),
+					},
+				},
+			},
+		},
+	}
+
+	cfg := config.New()
+
+	// test
+	c := Container(cfg, logger, otelcol)
+
+	// verify
+	assert.Equal(t, resource.MustParse("100m"), *c.Resources.Limits.Cpu())
```

Comment:
just a nit: asserting on the whole `corev1.ResourceRequirements` might be better. I would extract to to a variable and compare it here.